data/crac/crac-util/src/main/java/com/farao_community/farao/data/crac_util/CracAliasesUtil.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
+ */
+
+public final class CracAliasesUtil {
+
+    static final int NB_CHARACTER_IN_UCTE_NODE = 9;
+
+    private CracAliasesUtil() {
+    }
+
+    public static void createAliases(Crac crac, Network network) {
+        // List (without duplicates) all the crac elements that need to be found in the network
+        Set<String> elementIds = new HashSet<>();
+        crac.getCnecs().forEach(cnec -> elementIds.add(cnec.getNetworkElement().getId()));
+        crac.getContingencies().forEach(contingency -> handleAliases(contingency.getNetworkElements(), elementIds));
+        crac.getNetworkActions().forEach(networkAction -> handleAliases(networkAction.getNetworkElements(), elementIds));
+        crac.getRangeActions().forEach(rangeAction -> handleAliases(rangeAction.getNetworkElements(), elementIds));
+
+        // Try to find a corresponding element in the network, and add elementId as an alias
+        elementIds.forEach(elementId -> {
+            Optional<Identifiable<?>> correspondingElement = network.getIdentifiables().stream().filter(identifiable -> anyMatch(identifiable, elementId)).findAny();
+            correspondingElement.ifPresent(identifiable -> identifiable.addAlias(elementId));
+        });
+    }
+
+    private static void handleAliases(Set<NetworkElement> networkElements, Set<String> elementIds) {
+        networkElements.forEach(networkElement -> elementIds.add(networkElement.getId()));
+    }
+
+    /* It only works correctly for Cnec with direction = both. This corrupts the other Cnec.
+    TODO : only look for the elements with reverse ID if all the thresholds of the cnec have a BOTH direction.
+     */
+    private static boolean anyMatch(Identifiable<?> identifiable, String cnecId) {
+        return nameMatches(identifiable, cnecId, false) ||
+            aliasMatches(identifiable, cnecId, false) ||
+            nameMatches(identifiable, cnecId, true) ||
+            aliasMatches(identifiable, cnecId, true);
+    }
+
+    private static boolean nameMatches(Identifiable<?> identifiable, String cnecId, boolean reverse) {
+        return identifiable.getId().trim().matches(checkWithPattern(cnecId, reverse));
+    }
+
+    private static boolean aliasMatches(Identifiable<?> identifiable, String cnecId, boolean reverse) {
+        return identifiable.getAliases().stream().anyMatch(alias -> alias.trim().matches(checkWithPattern(cnecId, reverse)));
+    }
+
+    private static String checkWithPattern(String string, boolean reverse) {
+        int first = reverse ? NB_CHARACTER_IN_UCTE_NODE : 0;
+        int second = reverse ? 0 : NB_CHARACTER_IN_UCTE_NODE;
+        return Pattern.quote(string.substring(first, first + FIRST_7_CHARACTER_EQUAL.getNbCharacter())) + ".*" + " " + Pattern.quote(string.substring(second, second + FIRST_7_CHARACTER_EQUAL.getNbCharacter())) + ".*" + Pattern.quote(string.substring(BEGINNING_OF_ELEMENT_NAME.getNbCharacter())).trim();
```

Comment:
could you put a UcteNodeMatchingRule in argument of createAliases(), somehow transmit it to this method, and change the hard-coded FIRST_7_CHARACTER_EQUAL for the rule given by the user in argument of createAliases() ?

I want to force anyone using the createAliases method to explicitly specify a UcteNodeMatchingRule, to be sure that he knows what he is doing and that we do not create aliases that he doesn't expect :p

Reply:
I've done something, is that what you expected?

Reply:
Not exactly, here you always use FIRST_7_CHARACTER_EQUAL. And I would prefer that instead we use a UcteNodeMatchingRule given by the user.

Then you can do : 
CracAliasesUtil.createAliases(crac, network, UcteNodeMatchingRule.FIRST_7_CHARACTER_EQUAL)

And I can do (maybe one day in the RAO ^^) :
CracAliasesUtil.createAliases(crac, network, UcteNodeMatchingRule.ALL_8_CHARACTER_EQUAL)

I gives more flexibility to the feature and force the user the ask himself the question : which UcteNodeMatchingRule do I need.

Reply:
my bad, you already changed that, thank you !